sample_functions/sample_functions.py

```diff
@@ -51,29 +51,27 @@ def invoke(arg, from_tty):
         end_bbcount = int(args[1])
         interval = int(args[2])
 
-        current_bbcount = start_bbcount
+        function_p = re.compile(r'#[0-9]+  ([^\s]+) \(.*\) (at|from) .*')
 
-        function_p = re.compile(r'#0  0x[0-9a-f]+ in (\w+) .*')
+        # Save print address value so that we can restore it
+        print_address = gdb.parameter('print address')
+        gdb.execute('set print address off', to_string=True)
 
-        while current_bbcount <= end_bbcount:
+        for current_bbcount in range(start_bbcount, end_bbcount + 1, interval):
             udb.time.goto(current_bbcount)
-
-            # Get backtrace (including current function)
-            backtrace = gdb.execute('where', to_string=True)
-            backtrace = backtrace.splitlines()
-
-            line = backtrace[0]
-
-            m = function_p.match(line)
-
-            # Update current bbcount
-            current_bbcount = current_bbcount + interval
-
-            if m is None:
-                continue
-            function = m.group(1)
-
-            functions[function] += 1
+            frame = gdb.newest_frame()
+            # Create list of functions in the backtrace
+            trace_functions = []
+            while frame is not None:
+                trace_functions.append(frame.name())
```

Comment:
The name could be `None`. In this case maybe you could append it's pc (`str(frame.pc())`).

Reply:
Done